tools/cache_creator/cache_creator.cpp

```diff
@@ -161,6 +162,29 @@ bool hexStringToUuid(llvm::StringRef hexStr, llvm::MutableArrayRef<uint8_t> outU
   return true;
 }
 
+llvm::Expected<ElfLlpcCacheInfo> getCacheInfoFromMetadataBlob(llvm::StringRef noteBlob) {
+  llvm::msgpack::Document *document = new llvm::msgpack::Document;
```

Comment:
You can stack allocate this. LLVM only use `new` like this for IR objects that use a different memory management model.